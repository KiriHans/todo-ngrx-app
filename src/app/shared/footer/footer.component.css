
.information{
  display: flex;
  flex-direction: column;
  justify-content: center;

  font-size: 0.8rem;
  font-weight: 400;
  color: #292929;
  text-align: center;
}