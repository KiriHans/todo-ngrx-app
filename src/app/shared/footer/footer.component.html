<footer class="information">
  <p>Double-click to edit a todo</p>
  <p>Project created using Angular V16</p>
</footer>