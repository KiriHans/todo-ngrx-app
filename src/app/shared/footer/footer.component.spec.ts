import { ComponentFixture, TestBed } from '@angular/core/testing';

import { FooterComponent } from './footer.component';
import { DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';

describe('FooterComponent', () => {
  let component: FooterComponent;
  let fixture: ComponentFixture<FooterComponent>;
  let el: DebugElement;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [FooterComponent],
    });
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(FooterComponent);
    component = fixture.componentInstance;
    el = fixture.debugElement;
    compiled = fixture.nativeElement;

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  it('should contain a footer with class "information"', () => {
    const footer = el.query(By.css('.information'));

    expect(footer).toBeTruthy();
  });

  it('should contain basic information on how to add and edit todo elements', () => {
    const paragraphs = el.queryAll(By.css('p'));

    expect(paragraphs.length).toBe(3);

    const paragraphTextList: string[] = paragraphs.map((p) => {
      return p.nativeElement.innerHTML;
    });
    const editTodoParagraph = paragraphTextList.find((p) => p.includes('edit a todo'));
    const addTodoParagraph = paragraphTextList.find((p) => p.includes('add a new todo'));

    expect(editTodoParagraph).toBeTruthy();
    expect(editTodoParagraph).toContain('Double-click');

    expect(addTodoParagraph).toBeTruthy();
    expect(addTodoParagraph).toContain('Enter');
  });
});
