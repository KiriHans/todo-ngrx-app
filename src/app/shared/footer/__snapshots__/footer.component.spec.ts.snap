// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <footer
    class="information"
  >
    <p>
      <strong>
         Double-click 
      </strong>
       to edit a todo
    </p>
    <p>
      <strong>
         Enter 
      </strong>
       to add a new todo
    </p>
    <p>
      Project created using Angular V16
    </p>
  </footer>
</div>
`;
